addons/microsoft_calendar/models/calendar.py

```diff
@@ -60,8 +60,10 @@ def _restart_microsoft_sync(self):
     def _check_microsoft_sync_status(self):
         """
         Returns True if synchronization with Outlook Calendar is active and False otherwise.
+        The 'microsoft_synchronization_stopped' variable needs to be 'False' and Outlook account must be connected.
         """
-        return not self.env.user.microsoft_synchronization_stopped
+        outlook_connected = self.env.user._get_microsoft_calendar_token() and self.env.user._is_microsoft_calendar_valid()
+        return outlook_connected and self.env.user.microsoft_synchronization_stopped is False
```

Comment:
I think this can be done as `microsoft_synchronization_stopped` is guaranteed to be a boolean
```suggestion
    return outlook_connected and not self.env.user.microsoft_synchronization_stopped
```

Reply:
oh, I forgot that boolean fields can be true/false/none. it's ok then :slightly_smiling_face: